ArcMovies/Extensions/UIImageExtension.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  UIImageExtension.swift
+//  ArcMovies
+//
+//  Created by Yuri Saboia Felix Frota on 20/01/18.
+//  Copyright © 2018 YuriFrota. All rights reserved.
+//
+
+import UIKit
+import Nuke
+
+extension UIImageView {
+    func loadImage(withURL url: URL?, defaultImage: UIImage = #imageLiteral(resourceName: "NoPoster")) {
+        let activityIndicator = UIActivityIndicatorView()
+        activityIndicator.setupOn(view: self)
+        activityIndicator.startAnimating()
+        Manager.shared.cancelRequest(for: self)
+        self.image = nil
+        guard let imageURL = url else {
+            activityIndicator.stopAnimating()
+            self.image = defaultImage
+            return
+        }
+        Manager.shared.loadImage(with: imageURL) { (images) in
+            activityIndicator.stopAnimating()
+            guard let image = images.value else {
+                self.image = defaultImage
+                return
+            }
+            self.image = image
+        }
+    }
+    
+    func loadImageInCollectionCell(withURL url: URL?, collection: UICollectionView, indexPath: IndexPath, cell: UICollectionViewCell, defaultImage: UIImage = #imageLiteral(resourceName: "NoPoster")) {
+        let activityIndicator = UIActivityIndicatorView()
+        activityIndicator.setupOn(view: self)
+        activityIndicator.startAnimating()
+        Manager.shared.cancelRequest(for: self)
+        self.image = nil
+        guard let imageURL = url else {
+            activityIndicator.stopAnimating()
+            self.image = defaultImage
+            return
+        }
+        Manager.shared.loadImage(with: imageURL) { (images) in
+            activityIndicator.stopAnimating()
+            guard let image = images.value else {
+                self.image = defaultImage
+                return
+            }
+            if collection.indexPath(for: cell)?.row == indexPath.row {
```

Comment:
The loadImage from manager runs it's async tasks, image download, on a background thread, it returns to the main thread only when finished, because it has to change UI elements, and UI elements mustn't be changed on background threads.